### PR TITLE
Small refactor to make cluster.Spec smaller and more reusable

### PR DESCRIPTION
**Pull request by g-gaston**
*Description of changes:*
It had become a bit difficult to work with, we had added too many things and we came up with a bunch of usecases to read different manifests that didn't necessary require the `Spec`

* Move the low level logic to `files.Reader`
* Move the logic to read manifests (`Release`, `Bundles` and eks-d release) to `cluster.ManifestReader`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/615#" title="Author self-approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/615#" title="Author self-approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/615#issuecomment-964633724" title="Approved">taneyland</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston,taneyland]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/cluster/spec.go
Comment (vivek-koppuru) [inline pkg/cluster/spec.go:150]: Do we want the default as this? I realize this is just code being moved, but saying something like "eks-a", "n/a" might be more descriptive if this function is being invoked.

File: pkg/cluster/spec.go
Comment (vivek-koppuru) [inline pkg/cluster/spec.go:163]: What about "eks-a-cli" here? We aren't using these user agents yet right?

**Review by vivek-koppuru — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/615#" title="Author self-approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/615#issuecomment-964633724" title="Approved">taneyland</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/615#pullrequestreview-802061070" title="Approved">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston,taneyland,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/cluster/spec.go
Comment (g-gaston) [reply pkg/cluster/spec.go:150]: the `userAgent` method already adds the `eks-a` prefix
The first method argument is just the sub component name `eks-a-subcomponent`

File: pkg/cluster/spec.go
Comment (g-gaston) [reply pkg/cluster/spec.go:163]: good point, but same, the `userAgent` will add that

**Pull request merged**
